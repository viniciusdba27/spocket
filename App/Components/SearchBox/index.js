import React, { Component } from 'react'
import { View } from 'react-native'
import { PropTypes } from 'prop-types'
import { Input, Button } from 'react-native-elements'
import Icon from 'react-native-vector-icons/FontAwesome'
import { Helpers, Metrics, Colors } from 'App/Theme'

import Style from './styles.js'

class SearchBox extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: '',
    }
  }

  render() {
    const { onPress } = this.props
    const { value } = this.state
    return (
      <View>
        <View style={(Helpers.fillRow, Style.relative)}>
          <View>
            <Input
              inputContainerStyle={Style.input}
              containerStyle={Style.inputContainer}
              inputStyle={[Style.inputStyle, Metrics.horizontalPadding]}
              placeholder="Enter keywords..."
              errorStyle={Style.inputError}
              onChangeText={(text) => this.setState({ value: text })}
            />
          </View>
          <View style={Style.buttonWrapper}>
            <Button
              buttonStyle={Style.button}
              containerStyle={Style.buttonContainer}
              icon={<Icon name="search" size={15} color={Colors.white} />}
              title=""
              onPress={() => onPress(value)}
            />
          </View>
        </View>
      </View>
    )
  }
}

SearchBox.propTypes = {
  onPress: PropTypes.func,
}

export default SearchBox
