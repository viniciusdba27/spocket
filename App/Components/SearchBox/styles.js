import { StyleSheet } from 'react-native'
import { Colors } from 'App/Theme'

const BUTTON_WIDTH = 75

export default StyleSheet.create({
  button: {
    borderRadius: 0,
    height: '100%',
    paddingVertical: 0,
  },
  buttonContainer: {
    borderRadius: 0,
  },
  buttonWrapper: {
    height: '100%',
    position: 'absolute',
    right: 0,
    top: 0,
    width: BUTTON_WIDTH,
  },
  input: {
    borderColor: Colors.gray1,
    borderWidth: 2,
  },
  inputContainer: {
    paddingLeft: 0,
    paddingRight: BUTTON_WIDTH,
  },
  inputError: {
    height: 0,
    margin: 0,
  },
  inputStyle: {
    height: 0,
  },
  relative: {
    position: 'relative',
  },
})
