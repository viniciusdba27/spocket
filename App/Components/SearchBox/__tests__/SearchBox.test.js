import 'react-native'
import React from 'react'
import SearchBox from '../index'
import renderer from 'react-test-renderer'

jest.useFakeTimers()

it('renders correctly', async () => {
  const component = renderer.create(<SearchBox onPress={() => true} />).toJSON()
  expect(component).toMatchSnapshot()
})
