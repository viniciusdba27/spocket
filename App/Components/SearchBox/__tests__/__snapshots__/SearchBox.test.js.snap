// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "paddingLeft": 0,
            "paddingRight": 75,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#e1e1e1",
              "borderWidth": 2,
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Enter keywords..."
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "height": 0,
                "minHeight": 40,
                "paddingHorizontal": 15,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "height": 0,
              "margin": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 75,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 0,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 0,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "padding": 8,
                "paddingVertical": 0,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï€‚
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
